### ISSUE TITLE
Atomic metadata operations don't work with the group permission

### ISSUE THREAD

**Issue by mstfdkmn** [irods/python-irodsclient#361]
In case a user has `own` access only provided by the group permission, then this user cannot add metadata via the atomic operations.

check the group permission on a collection:

```
ils -A testgroup
/tempZone/home/testgroup:
        ACL - g:testgroup#tempZone:own
```
try to add metadata by `apply_atomic_operations` method:

```
import os
import ssl
from irods.session import iRODSSession
from irods.meta import iRODSMeta, AVUOperation

with iRODSSession(irods_env_file=env_file, **ssl_settings) as session:
    col = session.collections.get("/tempZone/home/testgroup")
    col = metadata.apply_atomic_operations( AVUOperation(operation='remove', avu=iRODSMeta('attr1','val1','unit1')),
                                       AVUOperation(operation='add', avu=iRODSMeta('attr3','val3')),
                                       AVUOperation(operation='add', avu=iRODSMeta('attr2','val2','unit2')),
                                       AVUOperation(operation='remove', avu=iRODSMeta('attr2','val2','unit2')) )
```
What is being seen/received:
```
...
irods.exception.CAT_NO_ACCESS_PERMISSION: None
```
Once the user has personal permission, he/she can work with atomic metadata operations.

```
ils -A testgroup
/tempZone/home/testgroup:
        ACL - userA#tempZone:own

```

**Comment by d-w-moore** [comment]
This is likely an iRODS core code limitation.  Checking into it....

**Comment by d-w-moore** [comment]
Also : `col = metadata.apply_atomic_operations` should be  `col.metadata.apply_atomic_operations` in the demo script given in the problem description, though that has nothing to do with the issue at hand :)

**Comment by d-w-moore** [comment]
Reproduced.  The iRODS server responds with `CAT_NO_ACCESS_PERMISSION`
---
Script  `atomic_avus_on_group_owned.py
````
import os
import ssl
from irods.session import iRODSSession
from irods.meta import iRODSMeta, AVUOperation

from irods.test.helpers import make_session
with make_session() as session:
    DIR = "/tempZone/home/public/dir"  # owned by some group of which current user is a member
    col = session.collections.get( DIR )
    col.metadata.apply_atomic_operations( AVUOperation(operation='remove', avu=iRODSMeta('attr1','val1','unit1')),
                                       AVUOperation(operation='add', avu=iRODSMeta('attr3','val3')),
                                       AVUOperation(operation='add', avu=iRODSMeta('attr2','val2','unit2')),
                                       AVUOperation(operation='remove', avu=iRODSMeta('attr2','val2','unit2')) )

```

Full error dump:
```
(py3) ~$ python atomic_avus_on_group_owned.py
Traceback (most recent call last):
  File "atomic_avus_on_group_owned.py", line 13, in <module>
    AVUOperation(operation='remove', avu=iRODSMeta('attr2','val2','unit2')) )
  File "/home/daniel/python-irodsclient/irods/meta.py", line 103, in apply_atomic_operations
    self._manager.apply_atomic_operations(self._model_cls, self._path, *avu_ops)
  File "/home/daniel/python-irodsclient/irods/manager/metadata_manager.py", line 157, in apply_atomic_operations
    self._call_atomic_metadata_api(request)
  File "/home/daniel/python-irodsclient/irods/manager/metadata_manager.py", line 164, in _call_atomic_metadata_api
    response = conn.recv()
  File "/home/daniel/python-irodsclient/irods/connection.py", line 128, in recv
    raise get_exception_by_code(msg.int_info, err_msg)
irods.exception.CAT_NO_ACCESS_PERMISSION: None

```

**Comment by d-w-moore** [comment]
@mstfdkmn No sweat, just putting it there for  purposes.  BTW our core code team seems to have fixed the problem on the way to upcoming releases post 4.2.11 : 
   - [4.2.12](https://github.com/irods/irods/commit/cac49c24435044cd83cdbca573ddfa3aef023c33) 
   - [4.3.0](https://github.com/irods/irods/commit/041590d90b19d5214d76423e9066250b626abed2)

**Comment by trel** [comment]
Ah, so closing this as related to https://github.com/irods/irods/issues/6191 makes sense?

**Comment by d-w-moore** [comment]
Yes, exactly. The commits exist, post 4.2.11 on 4-2-stable and pre-4.3.0 on main, addressing that issue, and I did confirm this week that the PRC will not receive the errors seen here when the fix is compiled into iRODS.

**Comment by trel** [comment]
Perfect.   Please close.

**Comment by d-w-moore** [comment]
Closing as this is a server side issue irods/irods#6191 which has been addressed for iRODS server releases to follow 4.2.11
